### PR TITLE
W-13215870: Propagate traceState in opentelemetry with the correspond…

### PR DESCRIPTION
**Pull request by fsgonz**
…ing ancestor tag

### CODE REVIEW

**Comment by fsgonz** [comment]
--validate

File: modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/exporter/impl/OpenTelemetrySpanExporterFactory.java [hunk 1]
```diff
@@ -50,6 +53,10 @@ public class OpenTelemetrySpanExporterFactory implements SpanExporterFactory, Di
   @Inject
   private SpanExporterConfiguration configuration;
 
+  @Inject
+  FeatureFlaggingService featureFlaggingService;
```
Comment (alopezbizon) [inline modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/exporter/impl/OpenTelemetrySpanExporterFactory.java:57]: Does it need an access level modifier?

File: tests/infrastructure/src/test/java/org/mule/test/infrastructure/SpanTestHierarchyTestCase.java [hunk 2]
```diff
@@ -80,6 +81,118 @@ public class SpanTestHierarchyTestCase extends AbstractMuleTestCase {
...
+        .addTraceStateKeyValueAssertion("key1", "value1")
+        .addTraceStateKeyValueAssertion("key2", "value2")
+        .addTraceStateKeyValueAssertion("key3", "value3")
+        .endChildren()
+        .endChildren();
+
+    spanTestHierarchy.assertSpanTree();
+  }
+
+  @NotNull
```
Comment (alopezbizon) [inline tests/infrastructure/src/test/java/org/mule/test/infrastructure/SpanTestHierarchyTestCase.java:158]: Is this annotation necessary?

Comment (fsgonz) [reply tests/infrastructure/src/test/java/org/mule/test/infrastructure/SpanTestHierarchyTestCase.java:158]: Just for the IDE in some cases. I'll remove it.

File: modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/exporter/impl/OpenTelemetryTraceIdUtils.java [hunk 3]
```diff
@@ -165,7 +170,8 @@ private static boolean parentSpanContextIsValid(InternalSpan parentInternalSpan)
     return false;
   }
 
-  public static Map<String, String> getContext(OpenTelemetrySpanExporter openTelemetrySpanExporter) {
+  public static Map<String, String> getContext(OpenTelemetrySpanExporter openTelemetrySpanExporter,
```
Comment (IvanAndresFritzler) [inline modules/tracing/mule-tracer-exporter-impl/src/main/java/org/mule/runtime/tracer/exporter/impl/OpenTelemetryTraceIdUtils.java:173]: I'm suggesting a naming change. Also, can we add JavaDoc? 
```suggestion
  public static Map<String, String> getDistributedTraceContext(OpenTelemetrySpanExporter openTelemetrySpanExporter,
```

**Review by IvanAndresFritzler — CHANGES_REQUESTED** [review]

File: modules/tracing/mule-tracer-api/src/main/java/org/mule/runtime/tracer/api/sniffer/CapturedExportedSpan.java [hunk 4]
```diff
@@ -77,4 +80,11 @@ public interface CapturedExportedSpan {
    * @return the end span nanos.
    */
   long getEndSpanEpochNanos();
+
+  /**
+   * @return a map containing the key/value corresponding corresponding to the trace state of the captured span.
```
Comment (sofiamorseletto) [inline modules/tracing/mule-tracer-api/src/main/java/org/mule/runtime/tracer/api/sniffer/CapturedExportedSpan.java:85]: typo: corresponding is repeated

File: tests/infrastructure/src/test/java/org/mule/test/infrastructure/SpanTestHierarchyTestCase.java [hunk 5]
```diff
@@ -80,6 +80,117 @@ public class SpanTestHierarchyTestCase extends AbstractMuleTestCase {
   @Rule
   public ExpectedException expectedException = none();
 
+
+  @Test
+  public void testWhenTraceStateKeyAssertedAndNotPresentAssertionShouldFail() {
```
Comment (sofiamorseletto) [inline tests/infrastructure/src/test/java/org/mule/test/infrastructure/SpanTestHierarchyTestCase.java:85]: We can rename this test to: testWhenTraceStateKeyIsExpectedButIsNotPresentAssertionShouldFail

**Comment by fsgonz** [comment]
--validate

**Review by IvanAndresFritzler — APPROVED** [review]

**Comment by fsgonz** [comment]
--validate

**Comment by fsgonz** [comment]
--validate

**Comment by fsgonz** [comment]
--validate

**Comment by fsgonz** [comment]
--validate

**Comment by fsgonz** [comment]
--validate

**Pull request merged**